### ISSUE TITLE
Ruby 1.31.0.pre1 compilation failure with source-only package

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#23593]
```
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
```

```
$ gcc --version
gcc (Debian 9.3.0-8) 9.3.0
```

Currently the following fails, which is new:

```
gem install grpc --platform ruby --prerelease # fetches 1.31.0.pre1
```

**Comment by veblush** [comment]
How to reproduce:

```
$ docker run -it --rm ruby:2.7 /bin/bash
# gem install grpc --platform ruby --prerelease # fetches 1.31.0.pre1
...
make: *** No rule to make target '/usr/local/bundle/gems/grpc-1.31.0.pre1/src/ruby/ext/grpc/objs/opt/third_party/re2/re2/bitstate.o', needed by '/usr/local/bundle/gems/grpc-1.31.0.pre1/src/ruby/ext/grpc/libs/opt/libre2.a'.  Stop.
```

It seems that it's caused by a new re2 dependency.

**Comment by veblush** [comment]
@donnadionne I think `re2` needs to be added to a dependency list of `ruby_gem` in `build_handwritten.yaml` like php and python targets.

**Comment by apolcyn** [comment]
Yep -- there are no re2 files in https://github.com/grpc/grpc/blob/7e7a48f863218f39a1767e1c2b957ca8e4789272/grpc.gemspec#L52, which is generated from https://github.com/grpc/grpc/blob/7e7a48f863218f39a1767e1c2b957ca8e4789272/templates/grpc.gemspec.template#L53

**Comment by donnadionne** [comment]
fix: https://github.com/grpc/grpc/pull/23599 
backport to 1.31.0: https://github.com/grpc/grpc/pull/23602

**Comment by jtattermusch** [comment]
It good that we have a fix now, but I'm surprised this wasn't caught by the ruby distribtests.
Looks like we are missing a distribtest for ruby source-only packages? Can we add one so this doesn't happen the next time?
(I know for python we do have source package distribtests and they've caught problems before).

**Comment by apolcyn** [comment]
> It good that we have a fix now, but I'm surprised this wasn't caught by the ruby distribtests.
Looks like we are missing a distribtest for ruby source-only packages? Can we add one so this doesn't happen the next time?
(I know for python we do have source package distribtests and they've caught problems before).

Just added a distribtest on master for this in #23639

**Comment by apolcyn** [comment]
this has been fixed

**Comment by jtattermusch** [comment]
Fixed on the release branch by #23602.
